test/expression_test.ts

```diff
@@ -92,4 +92,28 @@ describe('expressions', () => {
     expectTranslate('foo.bar;').to.equal('foo.bar;');
     expectTranslate('foo[bar];').to.equal('foo[bar];');
   });
+  it('should cast numeric values to int when bitwise operators are used for a binary expression',
+     () => {
+       expectTranslates({
+         'var z: number = x | y': 'num z = (x as int) | (y as int);',
+         'var z: number = x & y': 'num z = (x as int) & (y as int);',
+         'var z: number = x ^ y': 'num z = (x as int) ^ (y as int);',
+         'var z: number = x << y': 'num z = (x as int) << (y as int);',
+         'var z: number = x >> y': 'num z = (x as int) >> (y as int);'
+       });
+     });
+  it('should cast numeric values to int when a shorthand bitwise assignment expression is used',
+     () => {
+       expectTranslates({
+         'var y: number = (y |= x)': 'num y = (y = (y as int) | (x as int));',
+         'var y: number = (y &= x)': 'num y = (y = (y as int) & (x as int));',
+         'var y: number = (y ^= x)': 'num y = (y = (y as int) ^ (x as int));',
+         'var y: number = (y <<= x)': 'num y = (y = (y as int) << (x as int));',
+         'var y: number = (y >>= x)': 'num y = (y = (y as int) >> (x as int));',
+       });
+     });
+  it('should cast a number to int when evaluating bitwise not operators', () => {
```

Comment:
please remove this test, it's entirely redundant with the binary operator table test above.